docs/contextmanager.md

```diff
@@ -0,0 +1,67 @@
+The `KernelPool` context manager
```

Comment:
I am aware that the documentation should be written in rst. I feel more comfortable writing in markdown and expect some feedback regarding content, style, wording, etc. Once the final version is ready, I'll rewrite it in rst.

Reply:
okay sounds good, pandoc (https://pandoc.org/try/) should be able to handle it, but looking through this it already looks like rst to me

Reply:
I added the .rst version, and tested with sphinx and readthedocs. I didn't know about pandoc, but it also translates latex, html... great discovery!